Use (and pad) postage stamps on the edge of images.

TallJimbo opened the pull request:
I believe this addresses #2, but I haven't tested it at full scale yet.

TallJimbo committed:
Use (and pad) postage stamps on the edge of images.

esheldon commented:
I'll test it

esheldon commented:
It looks like, in cases where the edge is intersected, the image is sometimes not being copied in for the single-epoch image

esheldon commented:
this is true for the coadd as well

TallJimbo commented:
Can you get me an ID for an object with bad stamp(s)?

esheldon commented:
The very first object in the tile

![test-000000-bmask](https://cloud.githubusercontent.com/assets/621661/26631124/a4ac12b4-45d7-11e7-9f8e-db9e42a5b12b.png)
![test-000000-image](https://cloud.githubusercontent.com/assets/621661/26631127/aa9ba982-45d7-11e7-8e96-51fe615dbc18.png)


It turns out there are `NaN` in these.

esheldon commented:
The id for index 0 is 38553842652020741

TallJimbo commented:
The first object looks fine to me, and I'm pretty sure it's the same one from looking at the masks.

However, I am using `NaN` to in the part of the image where I have no data (in addition to setting the variance to `inf` and setting the `NO_DATA` mask plane); could that be messing up the scale range in your plots?  Would you prefer some other value for those regions?

esheldon commented:
yeah, at the end of one of my messages above I mentioned the NaN values were the problem

It is OK to use NaN there if you want, but I'll most likely convert them to zeros so that viewers won't go bonkers

esheldon commented:
It looks like the seg maps are not "shifted" to deal with the edge, here is an example

![test-000009-image](https://cloud.githubusercontent.com/assets/621661/26681254/76613906-46aa-11e7-9fa6-b79b824b0064.png)

![test-000009-bmask](https://cloud.githubusercontent.com/assets/621661/26681256/766231a8-46aa-11e7-9ef0-e2b97e796606.png)

![test-000009-seg](https://cloud.githubusercontent.com/assets/621661/26681255/7661f508-46aa-11e7-92a0-e1bac61cd2cb.png)

TallJimbo committed:
Pad segmaps when a stamp is on the edge, too

TallJimbo committed:
Remove To-Do item about stamps on edges, now that it's done

TallJimbo commented:
Should be fixed now.

esheldon committed:
factor out the configuration and propagate some config vars

Now there are default configs for both producer and maker
in defaults.py

propagate the min/max_box_size into the producer, adjusting
that logic appropriately

still using odd box sizes though....

esheldon commented:
Any idea what is going on with this one?  It looks like the bright neighbor to the top does not have a seg map entry

![test-segrad5-009074-image](https://cloud.githubusercontent.com/assets/621661/26694458/75e98ade-46d5-11e7-828d-73eb55c04926.png)

![test-segrad5-009074-bmask](https://cloud.githubusercontent.com/assets/621661/26694467/7b91677c-46d5-11e7-8d13-c6c8a6033ad2.png)

![test-segrad5-009074-seg](https://cloud.githubusercontent.com/assets/621661/26694470/7f18f1bc-46d5-11e7-9d24-4b7d7dffca6c.png)

TallJimbo commented:
I saw some missing bright objects in the catalog when I was making the segmaps, and I think there must have been a bug in the processing run, and I've seen some ticket traffic fly by since then that might have explained it (and fixed it).  Someone else at Princeton is currently reprocessing the same dataset with a more recent version of the pipeline; I'll update the directory the test code is pointing at when that's done and let you know.

esheldon commented:
ok, I'm going to go ahead an merge this.
